docs/proposals/ose-rebase.md

```diff
@@ -73,21 +228,54 @@ last_tag="$( git describe --abbrev=0 --tags )"
 #
 GIT_SEQUENCE_EDITOR=rebase.py git rebase -i upstream/master
 
+# Will be needed for creating the custom changelog for tito.
+CURRENT_HEAD=$(git merge-base master upstream/master)
+
 # We need to retag, because the rebase removed the latest tito tag.
 # Needs to be forced because while the tag is removed from the branch,
 # it still exists globally in the repository.
 git tag -f "${last_tag}" HEAD
+
+# Merge web-console stuff in
+VC_COMMIT="$(GIT_REF=enterprise-${OSE_VERSION} hack/vendor-console.sh 2>/dev/null | grep "Vendoring origin-web-console" | awk '{print $4}')"
+git add pkg/assets/bindata.go
+git add pkg/assets/java/bindata.go
+git commit -m "[DROP] Merge remote-tracking branch enterprise-${OSE_VERSION}, bump origin-web-console ${VC_COMMIT}"
+
+# Create custom changelog for tito
+declare -a changelog
+for commit in $( git log "${PREVIOUS_HEAD}..${CURRENT_HEAD}" --pretty=%h --no-merges ); do
+  changelog+=( "--changelog='$( git log -1 "${commit}" --pretty='%s (%ae)' )'" )
+done
+# Tag
+tito tag --accept-auto-changelog "${changelog[@]}"
+# Force-update master and push the latest tag
+git push origin ${CURRENT_BRANCH} -f refs/tags/${last_tag}
 ```
+### What changes for build cops
+
+Common conflicts can rise when new changes step on branding or tooling code. Manual
+resolution is the only thing we can do in such cases. Today when a conflict occurs, it is
+resolved manually ie. master is updated with the conflict resolved and a new build is
+kicked off. This is not going to be possible with the suggested process because the
+changelog for tito needs to be constructed based on the previous and the current HEAD of
+Origin that got pulled in with the rebase. All the steps above need to run manually and
```

Comment:
I guess we can script this out and require human intervention only during the conflict resolution.